src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -826,5 +826,106 @@ public static IEnumerable<MethodDesc> EnumAllVirtualSlots(MetadataType type)
                 } while (type != null);
             }
         }
+
+        /// <summary>
+        /// Try to resolve a given virtual static interface method on a given constrained type and its base types.
+        /// </summary>
+        /// <param name="constrainedType">Type to attempt virtual static method resolution on</param>
+        /// <param name="interfaceType">Interface declaring the method</param>
+        /// <param name="interfaceMethod">Interface method to resolve</param>
+        /// <param name="allowVariantMatches">True when variant matches are allowed</param>
+        /// <returns>MethodDesc of the resolved virtual static method, null when not found (runtime lookup must be used)</returns>
+        public override MethodDesc ResolveInterfaceMethodToStaticVirtualMethodOnType(MethodDesc interfaceMethod, TypeDesc interfaceType, TypeDesc currentType, bool allowVariantMatches)
```

Comment:
Would it make sense to follow the existing scheme instead of a bool flag? (We have `ResolveVariantInterfaceMethodToVirtualMethodOnType` and `ResolveInterfaceMethodToVirtualMethodOnType` instead of a bool flag for the non-static case.)

Reply:
Refactored in later commits.